pkg/check/datastore.go

```diff
@@ -52,18 +96,29 @@ func CheckStorageClasses(ctx *CheckContext) error {
 		for k, v := range sc.Parameters {
 			switch strings.ToLower(k) {
 			case dsParameter:
-				if err := checkDataStore(ctx, v, infra); err != nil {
+				if err := checkDataStore(ctx, v, infra, dsTypes); err != nil {
 					klog.V(2).Infof("CheckStorageClasses: %s: %s", sc.Name, err)
 					errs = append(errs, fmt.Errorf("StorageClass %s: %s", sc.Name, err))
 				}
 			case storagePolicyParameter:
-				if err := checkStoragePolicy(ctx, v, infra); err != nil {
+				if err := checkStoragePolicy(ctx, v, infra, dsTypes); err != nil {
 					klog.V(2).Infof("CheckStorageClasses: %s: %s", sc.Name, err)
 					errs = append(errs, fmt.Errorf("StorageClass %s: %s", sc.Name, err))
 				}
+			default:
+				// There is neither datastore: nor storagepolicyname: in the StorageClass,
+				// check the default datastore and collect its type.
+				if err := checkDefaultDatastoreWithDSType(ctx, dsTypes); err != nil {
+					klog.V(2).Infof("CheckStorageClasses: %s: %s", sc.Name, err)
+				}
 			}
 		}
 	}
+
+	for dsType, count := range dsTypes.getDataStoreTypeCount() {
+		dataStoreTypesMetric.WithLabelValues(dsType).Set(float64(count))
```

Comment:
nit: I think we should lowercase everything we get from vsphere APIs, so as not to introduce additional labels if something changes.

Reply:
Added lowercase coversion in `addDataStore()`, https://github.com/openshift/vsphere-problem-detector/pull/43/files#diff-4d023f9a9c662cddd2bcde080da1d94e60119d4f41fae7b286d368ebe9d3320fR60